jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -53,203 +49,258 @@ def __init__(self, input_models, reffiles=None, **pars):
-            badmasks.append(mask)
-
-        # Compute median of stack os images using BADMASKS to remove low weight
-        # values
-        median_image = median(resampled_sci, nlow=nlow, nhigh=nhigh,
-                              badmasks=badmasks)
-        # for i in range(len(resampled_sci)):
-        #    data = resampled_sci[i]
-        #    mask = badmasks[i]
-        return median_image
-
-    def blot_median(self, median_image):
-        """IFU-specific version of blot_median."""
-        cubeblot = blot_cube_build.CubeBlot(median_image, self.input_models)
-        cubeblot.blot_info()
-        blot_models, input_list_number = cubeblot.blot_images()
-        for j in range(len(blot_models)):
-            k = input_list_number[j]
-            self.blot_models[k].data += blot_models[j].data
-            self.blot_models[k].meta = blot_models[j].meta
-
-
-class ErrorWrongInstrument(Exception):
-    """ Raises an exception if the instrument is not MIRI or NIRSPEC
-    """
-    pass
+            detector_name = uq_det[idet]
+            opt_info = (kern_size[0], kern_size[1], threshold_percent,
+                        diffarr, minarr, normarr, minarr_norm)
+            opt_model = self.create_optional_results_model(opt_info)
+            opt_model.meta.filename = self.make_output_path(
+                basepath=self.input_models.meta.asn_table.products[0].name,
+                suffix=detector_name + '_outlier_output')
+            log.info("Writing out intermediate outlier file {}".format(opt_model.meta.filename))
+            opt_model.save(opt_model.meta.filename)
+
+        del diffarr
+
+        # store some information if the second flagging step is to be done. 
+        if ifu_second_check:
+            # store where the minarr is nan (neighbor pixels have nan so differences produces a nan)
+            nanminarr = np.isnan(minarr)
+            nanindx = np.where(nanminarr)
+
+        # Update DQ flag
+        for i in range(len(self.input_models)):
+            model = self.input_models[i]
+            sci = model.data
+            dq = model.dq
+
+            detector = model.meta.instrument.detector.lower()
+            # only use data from the same detector
+            if detector == uq_det[idet]:
+
+                # There could be a large number of pixels with a sci value of NaN
+                # but the dq flag of DO_NOT_USE has not been set.
+                # This can occur in Non-science regions of the detector.
+                check = np.where(
+                    np.logical_and(~np.bitwise_and(dq, dqflags.pixel['DO_NOT_USE']).astype(bool),
+                                   np.isnan(sci)))
+                log.debug("Number of pixels DQ was not set to DO_NOT_USE and Sci array was Nan{} ".
+                          format(len(check[0])))
+                # set all pixels with dq = DO_NOT_USE to have sci values of Nan
+                bad = np.where(np.bitwise_and(dq, dqflags.pixel['DO_NOT_USE']).astype(bool))
```

Comment:
Same here. You shouldn't need the `np.where`. Without it, `bad` will be a full-sized array of booleans (True/False), which can then be used as array indexes in the next line.